modules/tensor_mechanics/include/kernels/GeneralizedPlaneStrainOffDiag.h

```diff
@@ -0,0 +1,46 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef GENERALIZEDPLANESTRAINOFFDIAG_H
+#define GENERALIZEDPLANESTRAINOFFDIAG_H
+
+#include "Kernel.h"
+#include "RankTwoTensor.h"
+#include "RankFourTensor.h"
```

Comment:
use forward declarations here and use include only in the .C file. We've worked hard to keep the include statements to a minimum to improve our compile times.

```
class RankTwoTensor;
class RankFourTensor;
```